### PR TITLE
Make `Lifecycle.PER_EXECUTION` more realistic

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
`BatikTranscoderFuzzer` emits a warning on macOS that I will either get rid of or ignore.

**Comment by fmeum** [comment]
Stacked onto #868

File: src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java [hunk 1]
```diff
@@ -37,64 +45,202 @@
...
+    Class<?> testClass = extensionContext.getRequiredTestClass();
+    if (testClass.getEnclosingClass() != null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The test class %s is an inner class, which is not supported with"
+                  + " LifecycleMode.PER_EXECUTION. Either make it a top-level class or set"
+                  + " lifecycle = LifecycleMode.PER_TEST on @FuzzTest.",
+              testClass.getName()));
+    }
+    Constructor<?> constructor;
```
Comment (bertschneider) [inline src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java:231]: `constructor` could be looked up and returned completely in the try-catch-block, which might simplify the code a little bit.

File: src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java [hunk 2]
```diff
@@ -37,64 +45,202 @@
...
+    try {
+      constructor = testClass.getDeclaredConstructor();
```
Comment (bertschneider) [inline src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java:233]: Nothing for this PR but I wonder if a general switch to `MethodHandle`s would be beneficial.

**Review by bertschneider — APPROVED** [review]
Puh, what a PR! Great additions and simplifications, thx. 
Just some minor remarks, feel free to merge.

**Review by bertschneider — APPROVED** [review]

**Pull request merged**